src/System.Private.CoreLib/shared/System/IO/TextWriter.cs

```diff
@@ -527,6 +538,17 @@ public virtual Task WriteAsync(string value)
             tuple, CancellationToken.None, TaskCreationOptions.DenyChildAttach, TaskScheduler.Default);
         }
 
+        /// <summary>
+        /// Equivalent to WriteAsync(stringBuilder.ToString()) however it uses the
+        /// StringBuilder.GetChunks() method to avoid creating the intermediate string 
+        /// </summary>
+        /// <param name="value">The string (as a StringBuilder) to write to the stream</param>
+        public async virtual Task WriteAsync(StringBuilder value)
```

Comment:
Same, this should probably validate that value isn't null.  That'll need to be in a non-async part so that the exception propagates synchronously.